### PR TITLE
Bug 1618751 - Push health improved Parent Push handling

### PR DESCRIPTION
**Pull request by camd**
I opted to change the name of `Parent Push` to `Commit History`.  So it now shows not only the SHA of the Parent, but also the SHAs of all commits.  

I moved the `job-view` `RevisionList` and `Revision` classes to be in `/shared/` so I could reuse them here.  I had to modify them slightly and cleaned up some crufty `CSS` that was hanging around.  I also made it so that the little clipboad icon for copying the sha only shows on hover.

The second commit makes the updates to `Commit History`.  It tries one API the works in most cases.  But that API seems to fail for `autoland`, so I fall back to the original API in that case.

### CODE REVIEW

File: tests/push_health/test_compare.py [hunk 1]
```diff
@@ -25,28 +25,101 @@ def test_get_parent(test_push, test_repository):
         author='foo@bar.baz',
         time=datetime.datetime.now()
     )
+
+    autorel_commits = {'changesets': [
+        {
+            'author': 'Cheech Marin <cheech.marin@gmail.com>', 'backsoutnodes': [],
```
Comment (sarah-clements) [inline tests/push_health/test_compare.py]: lol nice...

File: tests/ui/job-view/revisions_test.jsx [hunk 2]
```diff
@@ -1,191 +1,195 @@
...
+  //     performance_alerts_enabled: true,
+  //     pushlogURL:
+  //       'https://hg.mozilla.org/integration/mozilla-inbound/pushloghtml',
+  //   });
+  //
+  //   mockData = {
+  //     revision,
+  //     repo,
+  //   };
+  // });
```
Comment (sarah-clements) [inline tests/ui/job-view/revisions_test.jsx]: Are you enabling this block in another pr?

File: ui/shared/Revision.jsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faUser } from '@fortawesome/free-regular-svg-icons';
+import { Row } from 'reactstrap';
+
+import { parseAuthor } from '../helpers/revision';
+
+import BugLinkify from './BugLinkify';
+import Clipboard from './Clipboard';
+
+export function Initials(props) {
```
Comment (sarah-clements) [inline ui/shared/Revision.jsx]: I think `AuthorInitials` would be a little clearer.

File: tests/ui/job-view/revisions_test.jsx (hunk 2 above)
Comment (camd) [reply tests/ui/job-view/revisions_test.jsx]: Oops, just forgot to delete it.  :)

File: ui/shared/Revision.jsx (hunk 3 above)
Comment (camd) [reply ui/shared/Revision.jsx]: Good call, yeah.  I'll fix it.

**Pull request merged**